lib/graphql/execution/lookahead.rb

```diff
@@ -87,16 +87,28 @@ def selected?
 
       # Like {#selects?}, but can be used for chaining.
       # It returns a null object (check with {#selected?})
+      # @param field_name [String, Symbol]
       # @return [GraphQL::Execution::Lookahead]
       def selection(field_name, selected_type: @selected_type, arguments: nil)
-        next_field_name = normalize_name(field_name)
+        next_field_defn = case field_name
+        when String
+          @query.get_field(selected_type, field_name)
+        when Symbol
+          # Try to avoid the `.to_s` below, if possible
+          all_fields = @query.warden.fields(selected_type)
+          if (match_by_orig_name = all_fields.find { |f| f.original_name == field_name })
+            match_by_orig_name
+          else
+            guessed_name = Schema::Member::BuildType.camelize(field_name.to_s)
```

Comment:
If all you want to avoid is the new allocation when doing `field_name.to_s`, doesn't this work:

```
guessed_name = Schema::Member::BuildType.camelize(-field_name.to_s)
```

Reply:
Thanks for taking a look, @rafaelfranca. I investigated and found that `-field_name.to_s` _does_ allocate a new string, although it then de-duplicates it with an existing frozen string in which is already in memory. So, it would help in a place where that string was retained, but it didn't prevent the initial allocation. 

However, I did find `Symbol#name` which reduces allocations here, I'll use that it #4212